staging/src/k8s.io/apimachinery/pkg/util/duration/duration_test.go

```diff
@@ -74,9 +74,14 @@ func TestHumanDurationBoundaries(t *testing.T) {
 		{d: 8 * 24 * time.Hour, want: "8d"},
 		{d: 8*24*time.Hour + 23*time.Hour, want: "8d"},
 		{d: 2*365*24*time.Hour - time.Millisecond, want: "729d"},
-		{d: 2 * 365 * 24 * time.Hour, want: "2y0d"},
-		{d: 2*365*24*time.Hour + 23*time.Hour, want: "2y0d"},
-		{d: 3 * 365 * 24 * time.Hour, want: "3y0d"},
+		{d: 2 * 365 * 24 * time.Hour, want: "2y"},
+		{d: 2*365*24*time.Hour + 23*time.Hour, want: "2y"},
+		{d: 2*365*24*time.Hour + 23*time.Hour + 59*time.Minute, want: "2y"},
```

Comment:
Check both sides of the boundary condition?

```
{d: 2*365*24*time.Hour + 24*time.Hour - 1, want: "2y"},
{d: 2*365*24*time.Hour + 24*time.Hour, want: "2y1d"}
```

Reply:
okay, should I do the same for the 8y boundary?

Reply:
oh, never mind. I just realized 8y boundary already does it

Reply:
The point is to demonstrate exactly where the cutoff is by having a case for each value. Looks like the boundary is not where I thought it was, judging from the 8y case.

Reply:
Oh, I see, the 8y case is a different code path completely.